scripts/mbedtls_dev/test_generation.py

```diff
@@ -0,0 +1,219 @@
+"""Common test generation classes and main function.
```

Comment:
I only just realized, but we already have test _code_ generation (`tests/scripts/generate_test_code.py`, turns `.function` files into `.c`), so “test generation” is ambiguous: we should specify that this is test _data_ generation. I propose (in a follow-up) to rename this module to `test_data_generation`.